### ISSUE TITLE
Lib partly misleading and incomplete

### ISSUE THREAD

**Issue by pmpfeil** [hypfvieh/paulmann-bt-lib#1]
As the manufacturer of the Paulmann BLE devices we like to public our API very soon. Please note that the information in this repository is partly misleading and incomplete.

**Comment by hypfvieh** [comment]
If you are really related to Paulmann, I will be glad to use the official API and retire this project or at least fix it as soon as the API is available.

Could you please provide a link or something to the official API?

As there was no public API at the time of writing this library, all feature etc. are based on observations, assumptions and debug output.

I never claimed that these observations and assumptions are either complete or correct at all.
Everything was tested with the hardware I got and was working for my use cases.

**Comment by pmpfeil** [comment]
Hello David,

Yes, we are the „real“ and as soon as we are ready to publish, I will let you know. We would be then happy, that you withdraw your information to avoid confusion. Meanwhile, if you have still projects running, please feel free to come back to me for more details. Especially the password is not for the remote; that’s two different cases. The password blocks any connection if not correct (default is 0123), whilst the remote might still get in contact via broadcasting. Remote characteristic is only to block 3d party remotes (e.g. neighbors), in case the user won’t have one own (only one remote per device can be used).

Further I like to mention that we feel partly affected with our copy rights, but we are not following up this issue, as we like to promote open innovation and usage of our products.

On the other hand we are searching some programmer for the BLE topic. Might this be interesting for you?

Viele Grüße / Kind regards

i.A. Torsten Pfeil
Sortimentsmanagement / Assortment Management

Fon +49 5041 998-199
Fax +49 5041 998-4199
tpfeil@paulmann.de<mailto:tpfeil@paulmann.de>

Paulmann Licht GmbH
Quezinger Feld 2
31832 Springe-Völksen
Deutschland
www.paulmann.com<http://www.paulmann.com/>

[facebook]<http://www.facebook.com/PaulmannLicht>   [youtube] <http://www.youtube.com/paulmannlicht>    [instagramm] <https://instagram.com/paulmann_licht/>    [googlePlus] <https://plus.google.com/u/0/b/102040436175585192049/102040436175585192049/about>

Sitz der Gesellschaft: Springe-Völksen
Amtsgericht Hannover HRB 101047
Geschäftsführer: Oliver Windbrake (Vorsitz), Thomas Hamann, Dirk Oldenburg



Von: David M. [mailto:notifications@github.com]
Gesendet: Freitag, 29. Juni 2018 10:19
An: hypfvieh/paulmann-bt-lib <paulmann-bt-lib@noreply.github.com>
Cc: Pfeil, Torsten <tpfeil@paulmann.de>; Author <author@noreply.github.com>
Betreff: Re: [hypfvieh/paulmann-bt-lib] Lib partly misleading and incomplete (#1)


If you are really related to Paulmann, I will be glad to use the official API and retire this project or at least fix it as soon as the API is available.

Could you please provide a link or something to the official API?

As there was no public API at the time of writing this library, all feature etc. are based on observations, assumptions and debug output.

I never claimed that these observations and assumptions are either complete or correct at all.
Everything was tested with the hardware I got and was working for my use cases.

**Comment by pmpfeil** [comment]
Hello David,

Yes, we are the „real“ and as soon as we are ready to publish, I will let you know. We would be then happy, that you withdraw your information to avoid confusion. Meanwhile, if you have still projects running, please feel free to come back to me for more details. Especially the password is not for the remote; that’s two different cases. The password blocks any connection if not correct (default is 0123), whilst the remote might still get in contact via broadcasting. Remote characteristic is only to block 3d party remotes (e.g. neighbors), in case the user won’t have one own (only one remote per device can be used).

Viele Grüße / Kind regards

i.A. Torsten Pfeil
Sortimentsmanagement / Assortment Management

**Comment by pmpfeil** [comment]
Yes, we are the „real“ and as soon as we are ready to publish, I will let you know. We would be then happy, that you withdraw your information to avoid confusion. Meanwhile, if you have still projects running, please feel free to come back to me for more details. Especially the password is not for the remote; that’s two different cases. The password blocks any connection if not correct (default is 0123), whilst the remote might still get in contact via broadcasting. Remote characteristic is only to block 3d party remotes (e.g. neighbors), in case the user won’t have one own (only one remote per device can be used).

**Comment by hypfvieh** [comment]
Great news!
I will take a look at the public API as soon as it is available and will update my projects. If I still have any questions left, I will come back to you.

**Comment by hypfvieh** [comment]
Any news on the official library?
Has been quiet a while since your last post ...

**Comment by pmpfeil** [comment]
Hello,

Yes, sorry – I’m leaving the company and so I stopped my activities to publish the API. Please find it attached for your private use. I would be happy to have your comments on it as I’m freshmen to this topic.

Best regards
Torsten

**Comment by hypfvieh** [comment]
Sorry, Github does not add attachments.... 
Could you upload it somewhere or send the attachment to my address?
(can be found [here](https://github.com/hypfvieh/dbus-java/blob/master/pom.xml#L153))

**Comment by ghost** [comment]
Is there a chance to get the adapter as well ?

**Comment by hypfvieh** [comment]
which adapter?
I don't know if I'm permitted to publish the document I received, so I won't share it for now ..

**Comment by ghost** [comment]
I am looking for the Paulmann BLE adapter for using with iobroker. Paulmann did not release its version as the responsible guy left Paulmann.

**Comment by hypfvieh** [comment]
I know that the guy left. 

He left me some documentation about the BLE light controllers. But I did not found any new information in his notes. It's basically the same stuff I already knew from debugging and analysing the android app.

So everything I know can be found in the sources in this project.

**Comment by ghost** [comment]
So there is no version available, which can be easily installed by me as a complete newbie for iobroker ?

**Comment by hypfvieh** [comment]
In short: No.

I don't even no what iobroker is supposed to be.
This is a library for developers to create software which will then communicate with paulmann light devices. If you are not a developer, this is not the library you want to use.
